### PR TITLE
Local affinity processing

### PR DESCRIPTION
**Pull request by rogeralsing**

### CODE REVIEW

File: src/Proto.Cluster/ClusterHeartBeat.cs [hunk 1]
```diff
@@ -36,7 +46,7 @@ public ClusterHeartBeat(Cluster cluster)
 
         public Task StartAsync()
         {
-            var props = Props.FromProducer(() => new ClusterHeartBeatActor());
+            var props = Props.FromProducer(() => new ClusterHeartBeatActor(_system));
```
Comment (rogeralsing) [inline src/Proto.Cluster/ClusterHeartBeat.cs]: Do we need _system here? `context.System` exists once we are in the receive method

File: src/Proto.Cluster/Member/LocalAffinityPropsExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        /// Adds middleware which causes remote traffic against the actor to potentially poison it.
+        /// Useful with local affinity strategy to move partitioned workloads to the right node, even after a re-balance
+        /// </summary>
+        /// <param name="props"></param>
+        /// <param name="relocationFactor">Chance the actor is poisoned on remote traffic, 0-1</param>
+        /// <param name="hasLocalAffinity">Predicate on message envelope, to have local affinity only on partitioned messages</param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentException"></exception>
+        public static Props WithPoisonOnRemoteTraffic(this Props props,
+            float relocationFactor = DefaultRelocationFactor, Predicate<MessageEnvelope>? hasLocalAffinity = null)
```
Comment (rogeralsing) [inline src/Proto.Cluster/Member/LocalAffinityPropsExtensions.cs]: ooh, this is very nice, per message affinity checks. 👍🏻

**Comment by rogeralsing** [comment]
This looks great!

File: src/Proto.Cluster/ClusterHeartBeat.cs (hunk 1 above)
Comment (mhelleborg) [reply src/Proto.Cluster/ClusterHeartBeat.cs]: Cleaned up :)

File: src/Proto.Cluster/Member/LocalAffinityPropsExtensions.cs (hunk 2 above)
Comment (mhelleborg) [reply src/Proto.Cluster/Member/LocalAffinityPropsExtensions.cs]: Nice to have if we have some low volume traffic from elsewhere :)

**Pull request merged**